osu.Game/Screens/Multi/DrawableRoomPlaylistItem.cs

```diff
@@ -223,11 +229,26 @@ protected override void LoadComplete()
                 base.LoadComplete();
 
                 State.BindValueChanged(stateChanged, true);
+                FinishTransforms(true);
             }
 
             private void stateChanged(ValueChangedEvent<DownloadState> state)
             {
-                this.FadeTo(state.NewValue == DownloadState.LocallyAvailable ? 0 : 1, 500);
+                switch (state.NewValue)
+                {
+                    case DownloadState.LocallyAvailable:
+                        // Perform a local query of the beatmap by beatmap checksum, and reset the state if not matching.
+                        if (beatmapManager.QueryBeatmap(b => b.MD5Hash == playlistItem.Beatmap.Value.MD5Hash) == null)
```

Comment:
As an aside - this is way out of scope here and also a separate consideration for web, but in the long *long* run we might want a more reliable hash algorithm here - md5 is broken and I'm told collisions are easily manufactured. (Whether a colliding file can pass as a valid `.osu` file - I don't know)

Reply:
Definitely. I believe this could already be done quite easily. Not sure when we'll get to doing this (probably quite low priority until we notice people doing it).